include/parametriccurves/polynomial.hpp

```diff
@@ -138,8 +138,14 @@ struct Polynomial :
     if((t < this->t_min || t > this->t_max)){ throw std::out_of_range("TODO");}
     const time_t& dt (t);
     point_t h = coefficients_.col(order_);
-    for(std::size_t i=order_-1; i>=0; i--)
-      h = dt*h + coefficients_.col(i);
+    std::size_t i=order_-1; bool ok=true;

```

Comment:
Could you please specify what warnings you were getting? I see no difference in the for and while loops.

Reply:
There is a side effect when i==0, because i is know an unsigned int it wrap around to the highest positive integer. Therefore the condition is still true and generates a segfault. With the while you make sure to stop when i==0 *after* performing the iteration.